cmd/cli/version/version.go

```diff
@@ -104,13 +107,71 @@ func (oV *VersionOptions) Run(ctx context.Context, cmd *cobra.Command) error {
 	if !oV.ClientOnly {
 		serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
 		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msgf("could not get server version")
-			return err
+			return fmt.Errorf("error running version: %w", err)
+		} else {
+			versions.ServerVersion = serverVersion
+			columns = append(columns, serverVersionColumn)
 		}
+	}
+
+	// Add additional information
+	versions.OperatingSystem = runtime.GOOS
+	versions.Architecture = runtime.GOARCH
+	versions.UserID = "some-user-id" // Replace this with your method to get UserID
 
-		versions.ServerVersion = serverVersion
-		columns = append(columns, serverVersionColumn)
+	var serverVersion string
+	if versions.ServerVersion != nil {
+		serverVersion = versions.ServerVersion.GitVersion
 	}
 
+	checkForUpdates(ctx, versions.ClientVersion.GitVersion, serverVersion, versions.OperatingSystem, versions.Architecture, versions.UserID)
+
 	return output.OutputOne(cmd, columns, oV.OutputOpts, versions)
 }
+
+type ServerResponse struct {
+	Version string `json:"version"`
+	Message string `json:"message"`
+}
+
+func checkForUpdates(ctx context.Context, currentClientVersion, currentServerVersion, os, arch, userID string) {
```

Comment:
Let's modify this method to return an error, that way callers can decide how to handle different error cases.